src/server/db_slice.h

```diff
@@ -10,9 +10,35 @@
 #include "server/common.h"
 #include "server/conn_context.h"
 #include "server/table.h"
+#include "util/fibers/fibers.h"
 
 namespace dfly {
 
+// TODO: Move this, use it in all places we have variant<string, string_view>
+class StringOrView {
+ public:
+  static StringOrView FromString(std::string s);
+  static StringOrView FromView(std::string_view sv);
+
+  StringOrView() = default;
+
+  StringOrView(const StringOrView& o);
+  StringOrView(StringOrView&& o);
+  StringOrView& operator=(const StringOrView& o);
+  StringOrView& operator=(StringOrView&& o);
+
+  bool operator==(const StringOrView& o) const;
+  bool operator!=(const StringOrView& o) const;
+
+  std::string_view view() const;
+
+ private:
+  void Set(std::string s, std::string_view sv);
+
+  std::string s_;
```

Comment:
nice. why not variant, btw?

Reply:
> TODO: Move this, use it in all places we have variant<string, string_view>
 
Why do we need a big custom class if a simple wrapper is enough?  🤷🏻‍♂️ we have something similar here:

https://github.com/dragonflydb/dragonfly/blob/a93ad4e86f2e2a80e9d952bbd991ea954f0940fe/src/server/table.h#L102-L118

Reply:
lol, this is WIP, I didn't mean for anyone to review it yet :)

Re/ variant: I wanted to see an implementation with both members, where `s_` is only used as a storage medium, and `sv_` is always used for inspecting the string. But I agree that `variant` is better (except for an added `if` in the `visit` path, but that's super unimportant and premature optimization).

Re/ big custom class - I'm not sure what you mean by big, but the `Key` you linked to is also a "custom class" :rofl: It's just a private class that's not usable outside of table, and is one of this things that should be replaced by a general utility.

Anyway, I moved the `StringOrView` class to a standalone file, took the `variant` approach like `Key` and moved table.h to use it.

But, as I said, this is still WIP, so unless you are really bored, I wouldn't look at this PR yet :nerd_face: